### DISCUSSION TITLE
How can I rewrite `SpyInstance` using `MockInstance`?

### DISCUSSION THREAD

**Discussion by rinarakaki** [vitest-dev/vitest#6998] [Q&A]
The [doc](https://vitest.dev/guide/migration) says:

> `SpyInstance` is removed in favor of `MockInstance`

but `MockInstance` is not a drop-in replacement of `SpyInstance`. How can I migrate correctly?

**Comment by sheremet-va** [comment]
On the same page it says how all mock functions and types are also changed with examples: https://vitest.dev/guide/migration#simplified-generic-types-of-mock-functions-e-g-vi-fn-t-mock-t

**Reply by acolchagoff** [reply]
This page says nothing about SpyInstance.

**Reply by sheremet-va** [reply]
https://v3.vitest.dev/guide/migration.html#simplified-generic-types-of-mock-functions-e-g-vi-fn-t-mock-t

**Reply by NateRobinsonAA** [reply]
@sheremet-va Your link includes an anchor that doesn't seem to exist on that page. I also cannot find any instance of the phrase "simplified generic" on the page.

@acolchagoff On the page linked, I took inspiration from the Types section (https://vitest.dev/guide/migration.html#types) and made the following code change successfully:

Before:
```typescript
let sortColumnSpy: jest.SpyInstance;
sortColumnSpy = vi.spyOn(component.sortColumnName, 'emit');
```

After:
```typescript
import { Mock } from 'vitest';

let sortColumnSpy: Mock;
sortColumnSpy = vi.spyOn(component.sortColumnName, 'emit');
```

**Reply by sheremet-va** [reply]
> @sheremet-va Your link includes an anchor that doesn't seem to exist on that page. I also cannot find any instance of the phrase "simplified generic" on the page.

<img width="905" height="701" alt="Screenshot 2025-12-10 at 18 50 27" src="https://github.com/user-attachments/assets/35de53ed-fb08-410d-84ad-9f1fd6547cd4" />
